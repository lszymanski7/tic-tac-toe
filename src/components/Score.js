import React from 'react'
import PropTypes from 'prop-types'

const Score = ({ inProgress, score, turn }) => {
    /* istanbul ignore next */

    // Player X's turn
    const xTurn = inProgress && turn === 'X' ? 'score__turn' : ''

    /* istanbul ignore next */

    // Player O's turn
    const oTurn = inProgress && turn === 'O' ? 'score__turn' : ''

    return (
        <div className="score">
            <div className="score__container">
                <p
                    className={xTurn}
                    data-testid="player1-caption"
                >
                    Player 1  [X]
                </p>
                <p data-testid="player1-score">{score.player1}</p>
            </div>
            <div className="score__container">
                <p data-testid="draw-caption">Draw</p>
                <p data-testid="draw-score">{score.draw}</p>
            </div>
            <div className="score__container">
                <p
                    className={oTurn}
                    data-testid="player2-caption"
                >
                    Player 2  [O]
                </p>
                <p data-testid="player2-score">{score.player2}</p>
            </div>
        </div>
    )
}

Score.propTypes = {
    inProgress: PropTypes.bool.isRequired,
    score: PropTypes.object.isRequired,
    turn: PropTypes.string.isRequired
}

export { Score as default }
