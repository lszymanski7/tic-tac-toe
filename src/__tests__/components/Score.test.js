import React from 'react'
import { render, screen } from '@testing-library/react'
import Score from '../../components/Score'

const props = {
    inProgress: true,
    score: {
        player1: 0,
        draw: 0,
        player2: 0
    },
    turn: 'X'
}

describe('Score', () => {
    it('should be rendered correctly.', () => {
        const { container } = render(<Score {...props} />)
        expect(container).toMatchSnapshot()
    })

    it('should have the correct captions.', () => {
        render(<Score {...props} />)
        const player1 = screen.getByTestId('player1-caption')
        const draw = screen.getByTestId('draw-caption')
        const player2 = screen.getByTestId('player2-caption')
        expect(player1).toHaveTextContent('Player 1 [X]')
        expect(draw).toHaveTextContent('Draw')
        expect(player2).toHaveTextContent('Player 2 [O]')
    })

    it(`should have a red caption when it's player X's turn.`, () => {
        render(<Score {...props} />)
        const player1 = screen.getByTestId('player1-caption')
        expect(player1).toHaveClass('score__turn')
    })

    it(`should have a red caption when it's player O's turn.`, () => {
        render(<Score {...props} turn="O" />)
        const player2 = screen.getByTestId('player2-caption')
        expect(player2).toHaveClass('score__turn')
    })

    it('should have the correct score (default).', () => {
        render(<Score {...props} />)
        const player1 = screen.getByTestId('player1-score')
        const draw = screen.getByTestId('draw-score')
        const player2 = screen.getByTestId('player2-score')
        expect(player1).toHaveTextContent('0')
        expect(draw).toHaveTextContent('0')
        expect(player2).toHaveTextContent('0')
    })

    it('should have the correct score.', () => {
        render(<Score {...props} score={{ player1: 4, draw: 2, player2: 1 }}/>)
        const player1 = screen.getByTestId('player1-score')
        const draw = screen.getByTestId('draw-score')
        const player2 = screen.getByTestId('player2-score')
        expect(player1).toHaveTextContent('4')
        expect(draw).toHaveTextContent('2')
        expect(player2).toHaveTextContent('1')
    })
})
