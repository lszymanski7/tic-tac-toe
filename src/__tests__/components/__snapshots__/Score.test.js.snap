// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score should be rendered correctly. 1`] = `
<div>
  <div
    class="score"
  >
    <div
      class="score__container"
    >
      <p
        class="score__turn"
        data-testid="player1-caption"
      >
        Player 1  [X]
      </p>
      <p
        data-testid="player1-score"
      >
        0
      </p>
    </div>
    <div
      class="score__container"
    >
      <p
        data-testid="draw-caption"
      >
        Draw
      </p>
      <p
        data-testid="draw-score"
      >
        0
      </p>
    </div>
    <div
      class="score__container"
    >
      <p
        class=""
        data-testid="player2-caption"
      >
        Player 2  [O]
      </p>
      <p
        data-testid="player2-score"
      >
        0
      </p>
    </div>
  </div>
</div>
`;
